pkg/kube/client.go

```diff
@@ -640,7 +640,7 @@ func createPatch(target *resource.Info, current runtime.Object) ([]byte, types.P
 
 	if isUnstructured || isCRD {
 		// fall back to generic JSON merge patch
-		patch, err := jsonpatch.CreateMergePatch(oldData, newData)
+		patch, err := jsonmergepatch.CreateThreeWayJSONMergePatch(oldData, newData, currentData)
```

Comment:
Due to our backwards compatibility guarantees, this change needs to be opt-in. We need to capture a command line bool flag (that can also have an environment variable). Based on that we need to use the old or new path. We need to use the old path by default.

In the Helm developer call we discussed when we could make it the default (i.e. Helm v4) and when we can start to talk about that. A todo to make it the default or only option in Helm v4 would be great.

Reply:
CLI flag and TODO have been added.